node/src/chain_spec.rs

```diff
@@ -314,6 +314,15 @@ fn testnet_genesis(
         .iter()
         .map(|(para_id, _genesis_data, _boot_nodes)| (*para_id, 1000))
         .collect();
+    let para_id_boot_nodes: Vec<_> = para_ids
+        .iter()
+        .map(|(para_id, _genesis_data, boot_nodes)| (*para_id, boot_nodes.clone()))
+        .collect();
+    let para_ids: Vec<_> = para_ids
```

Comment:
is this to filter those not having bootnodes? if so please right a comment

Reply:
No, it's because I removed the bootnodes from the registrar genesis config, now bootnodes are part of the data preservers genesis.